### PR TITLE
Fix complexity and duplicated code on scale management

### PR DESCRIPTION
**Pull request by stockiNail**
Fix complexity and duplicated code on scale management

### CODE REVIEW

**Review by kurkle — COMMENTED** [review]
Couple of nits.

I'd add some old fashioned unit tests too, but let me take a look at that later. I'd like to test out a system here (and later adopt to chart.js) where the unit tests are run first and fixtures only if those pass.

File: src/scale.js [hunk 1]
```diff
@@ -2,30 +2,35 @@ import {isFinite, valueOrDefault} from 'chart.js/helpers';
...
-    scale.min = range.min;
-  }
-  if (isFinite(range.max) &&
-		typeof scale.options.max === 'undefined' &&
-		typeof scale.options.suggestedMax === 'undefined') {
-    changed = scale.max !== range.max;
-    scale.max = range.max;
-  }
+  let changed = changeScaleLimit(scale, range, {limit: 'min', suggestedLimit: 'suggestedMin'});
+  changed = changeScaleLimit(scale, range, {limit: 'max', suggestedLimit: 'suggestedMax'}, changed);
```
[the remark this answers is not in the document]
Comment (stockiNail) [reply src/scale.js:6]: done

File: src/scale.js [hunk 2]
```diff
@@ -2,30 +2,35 @@ import {isFinite, valueOrDefault} from 'chart.js/helpers';
...
+  const {limit, suggestedLimit} = options;
+  if (isFinite(range[limit]) &&
+		typeof scale.options[limit] === 'undefined' &&
+		typeof scale.options[suggestedLimit] === 'undefined') {
+    const changed = scale[limit] !== range[limit];
+    scale[limit] = range[limit];
+    return changed;
+  }
+  return changedLimit;
+}
```
[the remark this answers is not in the document]
Comment (stockiNail) [reply src/scale.js]: > BTW, noticed the original code has a bug too, if min was changed, but max was not, changed would be false. Do the new tests detect that?

I didn't do a specific test for that but in the other PR I was using OR ||

**Comment by stockiNail** [comment]
> BTW, noticed the original code has a bug too, if min was changed, but max was not, changed would be false. Do the new tests detect that?

I didn't do a specific test for that but in the other PR I was using OR ||.

In my opinion, the change should be applied if min or max (or both) are changed.

**Comment by stockiNail** [comment]
Now it's:

```javascript
  let changed = changeScaleLimit(scale, range, 'min', 'suggestedMin');
  const changed = changeScaleLimit(scale, range, 'max', 'suggestedMax') || changed;
```

is the same of 

```javascript
  cont changed = changeScaleLimit(scale, range, 'min', 'suggestedMin') || changeScaleLimit(scale, range, 'max', 'suggestedMax');
```

or am I wrong?

**Comment by stockiNail** [comment]
> I didn't do a specific test for that but in the other PR I was using OR ||.

I have implemented the test case: `box/adjustScaleMax`.

**Review by kurkle — APPROVED** [review]

**Pull request merged**